### ISSUE TITLE
[Bug] Adding a feed is case sensitiv

### ISSUE THREAD

**Issue by math-GH** [FreshRSS/FreshRSS#6904]
### Describe the bug

I had added https://mastodon.social/tags/FreshRSS.rss before.

Then I added https://mastodon.social/tags/freshrss.rss (same URL but the last part has only lower cases).

It was possible.

### To Reproduce

1. Go to add a feed and add https://mastodon.social/tags/FreshRSS.rss
2. add https://mastodon.social/tags/freshrss.rss too
3. see that both feeds work

### Expected behavior

it should not be possible to add a feed twice if it is the same URL. 
It should not be case sensitive.

### FreshRSS version

1.23.3-dev (latest)

### Environment information

- Database version: Mysql 5.7
- PHP version: PHP 8.1
-Web server type: Apache

**Comment by jesskay** [comment]
URLs _are_ case sensitive, it's just that most sites in practice normalize case when interpreting a URL, so normalizing case on the FreshRSS side could potentially break sites that don't normalize case.

That said, I don't believe it's likely that two feeds would have the exact same URL save for casing, so for this specific case it might be sufficient for FreshRSS to detect that you're adding a feed that only differs in casing from an existing feed, and ask if you're really sure?

**Comment by Alkarex** [comment]
Indeed, while the URL part until the end of the domain is case-insensitive, the rest is case sensitive.
Furthermore, we removed a while ago the uniqueness constraint on feed URLs, due to the various problems it created, such as when feeds change address.
So indeed, some kind of warning with a confirmation could be nice (for instance in JavaScript in the UI while adding the feed), but otherwise not something I believe should be changed
